packages/components/src/SegmentedControl/SegmentedControlOption.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { PropsWithChildren, useId, useMemo } from "react";
+import { useSegmentedControl } from "./SegmentedControlProvider";
+
+interface SegmentedControlOptionProps extends PropsWithChildren {
+  /**
+   * The value of the option.
+   */
+  readonly value: string | number;
+  /**
+   * The label to appear beside the radio button.
+   */
+  readonly checked?: boolean;
+  readonly label?: string;
+  readonly onChange?: (valueIn: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+export function SegmentedControlOption({
+  value,
+  checked,
+  children,
+  label,
+  onChange,
+}: SegmentedControlOptionProps) {
+  const { selectedOption, handleChange } = useSegmentedControl();
+  const localChecked = useMemo(() => {
+    return checked ?? selectedOption === value;
+  }, [selectedOption, checked]);
+  const inputId = `${value.toString()}_${useId()}`;
+
+  return (
+    <>
+      <input
+        type="radio"
+        id={inputId}
+        checked={localChecked}
+        value={value}
+        onChange={handleChangeLocal}
+      />
+      <label htmlFor={inputId}>{children || label}</label>
```

Comment:
If this is an icon only we'll need devs to add visually hidden text to add accessibility. Conversely we could say children can be whatever but an optional `accessibilityLabel` prop can be used when it's just an icon as a fast track to setting an `aria-label.`

Reply:
Appreciate all the feedback Eric! The PR is very much a draft, and I communicated internally about all the work this would still need to go through before it was ready. I wanted to get a V0.01 over the fence.

I'll be sure that whomever picks up this torch makes sure your feedback is addressed. :)

Reply:
Good call out - `aria-label` prop added here 82b04af7bb42b8c1841d789323edbbf2e5350c97

Reply:
Might suggest limiting it to `children`. `label` would be redundant